llvm/lib/Target/Hexagon/MCTargetDesc/HexagonMCTargetDesc.cpp

```diff
@@ -420,68 +504,31 @@ static bool LLVM_ATTRIBUTE_UNUSED checkFeature(MCSubtargetInfo* STI, uint64_t F)
 
 namespace {
 std::string selectHexagonFS(StringRef CPU, StringRef FS) {
-  SmallVector<StringRef, 3> Result;
+  SmallVector<std::string> Result;
   if (!FS.empty())
-    Result.push_back(FS);
-
-  switch (EnableHVX) {
-  case Hexagon::ArchEnum::V5:
-  case Hexagon::ArchEnum::V55:
-    break;
-  case Hexagon::ArchEnum::V60:
-    Result.push_back("+hvxv60");
-    break;
-  case Hexagon::ArchEnum::V62:
-    Result.push_back("+hvxv62");
-    break;
-  case Hexagon::ArchEnum::V65:
-    Result.push_back("+hvxv65");
-    break;
-  case Hexagon::ArchEnum::V66:
-    Result.push_back("+hvxv66");
-    break;
-  case Hexagon::ArchEnum::V67:
-    Result.push_back("+hvxv67");
-    break;
-  case Hexagon::ArchEnum::V68:
-    Result.push_back("+hvxv68");
-    break;
-  case Hexagon::ArchEnum::V69:
-    Result.push_back("+hvxv69");
-    break;
-  case Hexagon::ArchEnum::V71:
-    Result.push_back("+hvxv71");
-    break;
-  case Hexagon::ArchEnum::V73:
-    Result.push_back("+hvxv73");
-    break;
-  case Hexagon::ArchEnum::V75:
-    Result.push_back("+hvxv75");
-    break;
-  case Hexagon::ArchEnum::V79:
-    Result.push_back("+hvxv79");
-    break;
-
-  case Hexagon::ArchEnum::Generic: {
-    Result.push_back(StringSwitch<StringRef>(CPU)
-                         .Case("hexagonv60", "+hvxv60")
-                         .Case("hexagonv62", "+hvxv62")
-                         .Case("hexagonv65", "+hvxv65")
-                         .Case("hexagonv66", "+hvxv66")
-                         .Case("hexagonv67", "+hvxv67")
-                         .Case("hexagonv67t", "+hvxv67")
-                         .Case("hexagonv68", "+hvxv68")
-                         .Case("hexagonv69", "+hvxv69")
-                         .Case("hexagonv71", "+hvxv71")
-                         .Case("hexagonv71t", "+hvxv71")
-                         .Case("hexagonv73", "+hvxv73")
-                         .Case("hexagonv75", "+hvxv75")
-                         .Case("hexagonv79", "+hvxv79"));
-    break;
-  }
-  case Hexagon::ArchEnum::NoArch:
-    // Sentinel if -mhvx isn't specified
-    break;
+    Result.push_back(FS.str());
+
+  if (EnableHVX != Hexagon::ArchEnum::NoArch) {
+    std::string HVXFeature;
+    if (EnableHVX == Hexagon::ArchEnum::Generic) {
+      // Set the default HVX version for a given processor if -mhvx option with
+      // no value is specified.
```

Comment:
There's no `-mhvx` command-line option for LLVM.

LLVM should be frontend independent, so shouldn't describe things in terms of how one particular frontend works.  i.e. -- don't describe LLVM features in terms of how clang happens to behave.